src/messageComposer/middleware/textComposer/applyCommandSettings.ts

```diff
@@ -0,0 +1,55 @@
+import type { Middleware } from '../../../middleware';
+import type { TextComposerMiddlewareExecutorState } from './TextComposerMiddlewareExecutor';
+import type { CommandSuggestion } from './types';
+
+export type ApplyCommandSettingsMiddleware = Middleware<
+  TextComposerMiddlewareExecutorState<CommandSuggestion>,
+  'onChange' | 'onSuggestionItemSelect'
+>;
+
+export const createApplyCommandSettingsMiddleware =
+  (): ApplyCommandSettingsMiddleware => ({
+    handlers: {
+      onChange: ({ complete, forward, state }) => {
+        const { command } = state;
+
+        if (!command) {
+          return forward();
+        }
+
+        const trigger = `/${command.name}`;
+        const newText = state.text.replace(new RegExp(`^${trigger}(\\s|$)`), '');
```

Comment:
this does not trim white space following the command.

Reply:
If you copy-past "/ban   ", then the whitespaces are not trimmed. I think we should remove them here.

Reply:
This one will now work.